precompiles/auctions/tx.go

```diff
@@ -0,0 +1,126 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package auctions
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	cmn "github.com/evmos/evmos/v19/precompiles/common"
+	"github.com/evmos/evmos/v19/x/evm/core/vm"
+)
+
+const (
+	// BidMethod defines the ABI method name for the auctions
+	// Bid transaction.
+	BidMethod = "bid"
+	// DepositCoinMethod defines the ABI method name for the auctions
+	// DepositCoin transaction.
+	DepositCoinMethod = "depositCoin"
+)
+
+// Bid bids on the current auction with a specified Evmos amount that must be higher than the highest bid.
+func (p *Precompile) Bid(
+	ctx sdk.Context,
+	origin common.Address,
+	contract *vm.Contract,
+	stateDB vm.StateDB,
+	_ *abi.Method,
+	args []interface{},
+) ([]byte, error) {
+	sender, msgBid := NewMsgBid(args)
+
+	var (
+		// isCallerOrigin is true when the contract caller is the same as the origin
+		isCallerOrigin = contract.CallerAddress == origin
+		// isCallerSender is true when the contract caller is the same as the sender
+		isCallerSender = contract.CallerAddress == sender
+	)
+
+	// The provided sender address should always be equal to the origin address.
+	// In case the contract caller address is the same as the sender address provided,
+	// update the sender address to be equal to the origin address.
+	// Otherwise, if the provided sender address is different from the origin address,
+	// return an error because is a forbidden operation
+	if isCallerSender {
+		sender = origin
```

Comment:
should we update here the sender on the `msgBid` too?
Because we're executing the msg `_, err := p.auctionsKeeper.Bid(ctx, msgBid)`, and if we changed the sender, then the `p.SetBalanceChangeEntries(cmn.NewBalanceChangeEntry(sender, msgBid.Amount.Amount.BigInt(), cmn.Sub))` will be a different address

Reply:
Why are we changing the sender with the origin instead of checking if sender is authorized by origin?